patches/server/0896-fix-EndDragonFight-previouslyKilled-should-be-false-.patch

```diff
@@ -0,0 +1,20 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: elmital <54907162+elmital@users.noreply.github.com>
+Date: Tue, 19 Apr 2022 15:14:00 +0200
+Subject: [PATCH] fix: EndDragonFight previouslyKilled should be false for
+ newly created worlds
+
+
+diff --git a/src/main/java/net/minecraft/world/level/dimension/end/EndDragonFight.java b/src/main/java/net/minecraft/world/level/dimension/end/EndDragonFight.java
+index ed6ff35a2112cea4faa6e1458d1effc40fce5dd2..cddd14d8642de7c53c68ccc792d4775abb1c24b6 100644
+--- a/src/main/java/net/minecraft/world/level/dimension/end/EndDragonFight.java
++++ b/src/main/java/net/minecraft/world/level/dimension/end/EndDragonFight.java
+@@ -113,7 +113,7 @@ public class EndDragonFight {
+             }
+         } else {
+             this.dragonKilled = true;
+-            this.previouslyKilled = true;
++            this.previouslyKilled = nbt.contains("PreviouslyKilled") && nbt.getBoolean("PreviouslyKilled"); // Paper - Should be false for newly created worlds
```

Comment:
Any reason why you can't just make this false? It looks the only way this could happen is if the entire nbt compound is empty. However since needsStateScanning should be true, it'll cause the dragon to correctly be detected and reset to the correct values.

Reply:
I figured out the same as you but in the case I've missed a possibility that the nbt contains this key or if it could be for any other future/current reason it's handled. So for me it's just the safer way but can modify if well confirmed that is not needed.

Reply:
I guess for future proofing. But I mean, the only way for this to happen currently is if you manually remove it from the nbt.

Reply:
Just waiting for a Paper team feedback(thx for yours) and I'll apply the necessary changes and a rebase if it's all good.

Reply:
Reconsidering what you said I've made changes after more check and test with stateScanning option I changed the dragonKilled field too. I hope I didn't screw up with git, I know 1.19 is the priority but at least it's done ;)